pkg/node/node.go

```diff
@@ -474,34 +595,132 @@ func (n *Node) monitorKeepPublicKeySubmission(
 	}
 
 	defer subscriptionConflictingPublicKey.Unsubscribe()
-	defer subscriptionPublicKeyPublished.Unsubscribe()
 
-	select {
-	case event := <-publicKeyPublished:
-		logger.Infof(
-			"public key [%x] has been accepted by keep: [%s]",
-			event.PublicKey,
-			keepAddress.String(),
-		)
-	case event := <-conflictingPublicKey:
-		logger.Errorf(
-			"member [%x] has submitted conflicting public key for keep [%s]: [%x]",
-			event.SubmittingMember,
-			keepAddress.String(),
-			event.ConflictingPublicKey,
-		)
-	case <-monitoringCtx.Done():
-		logger.Warningf(
-			"monitoring of public key submission for keep [%s] "+
-				"has been cancelled: [%v]",
-			keepAddress.String(),
-			monitoringCtx.Err(),
-		)
-	case <-abort:
-		logger.Warningf(
-			"monitoring of public key submission for keep [%s] "+
-				"has been aborted",
-			keepAddress.String(),
-		)
+	pubkeyChecksCounter := 0
+	const maxPubkeyChecksCount = 3
+	const pubkeyCheckTick = 10 * time.Minute
+
+	pubkeyCheckTicker := time.NewTicker(pubkeyCheckTick)
+	defer pubkeyCheckTicker.Stop()
+
+	for {
+		select {
+		case event := <-conflictingPublicKey:
+			// Even in the case of a fork, a transaction with a conflicting
+			// public key submitted to the chain had to be signed with the
+			// operator's pubkey. For an honest operator, this situation should
+			// never happen, and if it happened, it is better to exit the
+			// monitoring immediately and do not re-attempt to submit public
+			// key for this node. It is clear that something is wrong with the
+			// operator that published the conflicting key and it is safer to
+			// abandon this keep.
+			logger.Errorf(
+				"member [%x] has submitted conflicting public key for keep [%s]: [%x]",
+				event.SubmittingMember,
+				keepAddress.String(),
+				event.ConflictingPublicKey,
+			)
+			return
+		case <-pubkeyCheckTicker.C:
```

Comment:
`const monitorKeepPublicKeySubmissionTimeout = 30 * time.Minute` on the top of the file is no longer used.

Reply:
Already addressed in the next PR: https://github.com/keep-network/keep-ecdsa/pull/616